sdk/storage/src/data_lake/clients/data_lake_client.rs

```diff
@@ -95,20 +29,17 @@ impl DataLakeClient {
         bearer_token: String,
         custom_dns_suffix: Option<String>,
         options: ClientOptions<DataLakeContext>,
-    ) -> Result<Arc<Self>, url::ParseError> {
+    ) -> Self {
```

Comment:
This now follows the pattern found in CosmosDB: no Result, no Arc, no ParseError